docs/RELEASE_NOTES.md

```diff
@@ -26,13 +26,21 @@
 
 7. Add Migration guide section which specifies explicitly if there are breaking changes and how to tackle them.
 
+### 2.4.0 - MMM DD, 2020
```

Comment:
We could decide that the library version should be 3.0.0 with that fix, lets' discuss in this thread.

Reply:
I don't think it's a big change for the library. It's just a dependency change. Or does the user of the library needs to do sth?

Reply:
Yes, when you check the integration tests, I have updated the workarounds for the state, if the library user does not change it, product sync might fail.

Reply:
I am assuming this is not big too, so not sure the best way for java-sync, it's clear in terms of semantic versioning and it should be 3.0.0, because, it is a breaking change..

Reply:
So is it a breaking change or not? It's written here `you don't need to provide a key field in the id field of the Reference. ` - so if you keep it the old way, it will work? Maybe describe a bit more what needs to be changed?

Reply:
I am not sure the best way for java-sync, it's clear in terms of semantic versioning and it should be 3.0.0, because, it is a breaking change.  We need to write a migration note for this, probably not in release notes but in the release tag's description.

Reply:
What about we wait for the shopping list and then release it as a new version 3.0.0? 😄

Reply:
haha, it makes sense :) But the discussion is a generic thing, maybe it's obvious, just increase the major as suggested by semver.

>Major version X (X.y.z | X > 0) MUST be incremented if any backwards incompatible changes are introduced to the public API. It MAY also include minor and patch level changes. Patch and minor version MUST be reset to 0 when major version is incremented.

Reply:
Yep, I think it's ok to increase a version to 3.0.0 here if we follow those rules you posted.